controllers/platform/services/services.go

```diff
@@ -358,14 +371,26 @@ func (j JobServiceHandler) MergeContainerSpec(containerSpec *corev1.Container) (
 	return c, err
 }
 
+// hasPostgreSQLConfigured returns true when either the SonataFlow Platform PostgreSQL CR's structure or the one in the Job service specification is not nil
+func (j JobServiceHandler) hasPostgreSQLConfigured() bool {
+	return (j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSQL != nil) ||
+		(j.platform.Spec.Persistence != nil && j.platform.Spec.Persistence.PostgreSQL != nil)
+}
```

Comment:
i wonder if this should be enhanced by adding `j.IsServiceSetInSpec` to the check as well. like this?
```golang
	return (j.IsServiceSetInSpec && j.platform.Spec.Services.DataIndex.Persistence != nil && j.platform.Spec.Services.DataIndex.Persistence.PostgreSQL != nil) ||
		(j.platform.Spec.Persistence != nil && j.platform.Spec.Persistence.PostgreSQL != nil)
```

Reply:
Will add